common-pg/src/main/scala/com/socrata/pg/store/RollupManager.scala

```diff
@@ -337,4 +358,22 @@ object RollupManager {
      SecondaryRollupInfo(rollupInfo.name.underlying, rollupInfo.soql)
 
   def shouldMaterializeRollups(stage: LifecycleStage): Boolean = stage == LifecycleStage.Published
+
+  def collectTableNames(selects: BinaryTree[Select]): Set[String] = {
+    selects match {
+      case Compound(_, l, r) =>
+        collectTableNames(l) ++ collectTableNames(r)
+      case Leaf(select) =>
+        select.joins.foldLeft(select.from.map(_.name).filter(_ != TableName.This).toSet) { (acc, join) =>
```

Comment:
so this only collects joined table names?

Reply:
it processes join queries too but not udf.